crossing-repair.lic

```diff
@@ -33,6 +87,77 @@ class CrossingRepair
     fput('sort auto head') if settings.sort_auto_head
   end
 
+  def repair_prep(town, settings)
+    # This method is designed to prep for repair by checking if we have a bank specified for the
+    # town in base-town.yaml, and checking if we have leather and/or metal items to repair and if
+    # there is a requisite metal or leather repair person specified in base-town.yaml.
+    # If we don't have the proper specifications in base-town.yaml, warn user, direct to submit issue.
+    # Returns: the hash of items to repair at either metal/leather repairer.
+
+    DRC.message("Validating bank and repairers...") if $debug_mode_crossing_repair
+
+    # Populate a hash of leather items to repair, and metal items to repair, based on user's
+    # yaml list of equipment. While having to repair different items at leather vs. metal
+    # repair is deprecated, we still need to support the legacy settings.
+    leather_items_to_repair = @equipment_manager.items.select(&:leather).reject(&:skip_repair)
+    metal_items_to_repair = @equipment_manager.items.reject(&:leather).reject(&:skip_repair)
+
+    # No bank specified in base-town.yaml for specified repair town? Message user. This method
+    # both withdraws and deposits coins, so having a bank specified is requisite.
+    unless @hometown_data['deposit']
```

Comment:
Should this check be ignored if `settings.sell_loot_skip_bank: true`?

Reply:
On second thought, I think this is required because `repair_withdrawal_amount` is required?

Reply:
Correct. This check is needed because the script calls `DRCM.ensure_copper_on_hand(@repair_withdrawal_amount, settings)` before the repair. We also, then, deposit leftover coin (with an yaml option to skip that). I _did_ consider checking our current wealth against `repair_withdrawal_amount`, checking if we also intend to skip coin deposit and if so ignore the pre-repair bank check. Might still roll that in.